CONTRIBUTING.md

```diff
@@ -0,0 +1,256 @@
+
+A comprehensive description of the bug or feature is essential. It helps the team to perform thorough testing of the solution, to include of pertinent information in the release notes, and for other relevant details. 
+
+Additionally, linking your future pull request to the corresponding issue with a detailed description will facilitate the development process and help reviewers to understand your solution. 
+
+Also, the Jira issue number is a required part of a branch name for your new code.
+
+To create Jira issue, please open the  [ LINK ](https://perconadev.atlassian.net/jira/software/c/projects/PS/issues) and
+authenticate with your favorite social network account (Google, Facebook, etc.). Then follow the instructions and verify
+your e-mail. If Jira starts to redirect you back and forth (their bug, not ours), just open a new browser window with
+the link above.
+
+Now you're ready to an create issue. Find the big "Create" button in the top menu and press it. Choose issue type from
+the second drop-down, saying it is a Bug, an Improvement, or a New feature. Fill in all the required fields below. Please, add accurate and detailed description.
+Good description is a big first step to the solution. The
+rest of the fields are mostly for internal use; the core team developers will fill it later.
+  
+  As soon as Jira issue is submitted, you are ready to start coding.
+  
+## Working with the Source Code
+
+### Forking code to your GitHub account.
+
+First, you should have a GitHub account. You can not work with the Percona repository directly, you have
+to [ "Fork" ](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo)
+our repository to your GitHub account. To do so,
+open [ Percona PS repository](https://github.com/percona/percona-server) in a browser. Then, right to the repository
+name you'll find the "Fork" button. Press it.
+Then return to your GitHub home page, find your "percona-server" repository and clone it locally.
+Note
+that [ creating SSH keys and putting public one to the GitHub ](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
+settings will save a lot of time for you with GitHub workflow.
+
+### Before you start coding 
+
+In the cloned local repository checkout the "trunk" branch you are going to work on. For 8.0.X releases is 8.0. For
+8.4.X series it is 8.4, and for 9.0.X it is "trunk".
+
+Now you are ready to create a branch for your contribution. Please name the branch according to the following pattern:
+
+```
+PS-9876-8.0-bug_in_some_module
+^^^^^^^
+Jira issue number
+        ^^^
+        Base version
+            ^^^^^^^^^^^^^^^^^^^
+            Very short definition of the issue 
+``` 
+
+The next step is to compile the code in your new branch before making changes.
```

Comment:
Should have a heading for each step - fork, compile, and so on.

Reply:
I moved this text under the heading, where it belongs to.